[Chu Thi Thanh] iP

lilythchu opened the pull request:
### DukePro
>"Your mind is for having ideas, not holding them." - David Allen ([source](https://pages.github.com/))

DukePro frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,
1. download it from [here](https://github.com)
2. double-click it
3. add your tasks.
4. let it manage your tasks for you :wink:

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to pratice Java too. Here's the <span style="color:red">`main`</span> method:
```
public class Main {
  public static void main(String[] args) {
    Application.launch(MainApp.class, args);
  }
}
```

Commit:
Add Gradle support

lilythchu committed:
L1:Great_Echo_Exit

lilythchu committed:
L2: Add, List

lilythchu committed:
L3: Mark as Done

lilythchu committed:
L4: ToDos, Events, Deadlines

lilythchu committed:
Automated Text UI Testing

lilythchu committed:
Handle Errors

lilythchu committed:
Delete

lilythchu committed:
Use Enums

lilythchu committed:
Lv7: Save

lilythchu committed:
Merge branch 'branch-Level-7'

lilythchu committed:
Lv8: Dates and Times

lilythchu committed:
Update testing for date format

lilythchu committed:
Merge branch 'branch-Level-8' and resolve conflict

lilythchu committed:
A-MoreOOP

lilythchu committed:
Create packages

lilythchu committed:
update path in runtest

lilythchu committed:
Merge branch 'add-gradle-support'

lilythchu committed:
configuring checkstyle

lilythchu committed:
Add JUnit tests

lilythchu committed:
Enable OS file saving

lilythchu committed:
Add JavaDoc comments

lilythchu committed:
Merge branch 'branch-A-JavaDoc'

lilythchu committed:
comply with a coding standard

lilythchu committed:
find tasks by searching keyword

lilythchu committed:
update ui test for find command

lilythchu committed:
Merge branch 'branch-A-CodingStandard' and resolve cmt conflicts.

lilythchu committed:
Merge branch 'branch-Level-9' and resolve cmt conflict.

lilythchu committed:
fix checkstyle errors

lilythchu committed:
Set up JavaFX

lilythchu committed:
create GUI

lilythchu committed:
rounded text box and circle avatar

lilythchu committed:
Add header

PangKuangWei commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,88 @@
+     * Constructor for Storage.
+     */
+    public Storage() {
+        this.filePath = "data" + File.separator + "lilychat.txt";
+        this.dirPath = "data";
```
The "Java coding standard (all)" page says that we should use the `this` keyword only when a field is shadowed by a method or constructor parameter to reduce unnecessary noise.

PangKuangWei commented on src/main/java/duke/UI.java:
```diff
@@ -0,0 +1,99 @@
+/**
+ * Represents the UI of Duke.
+ */
+public class UI {
```
I am unsure whether this is a problem but I would recommend changing this to "Ui" because `UI` could be mixed up with constants. The example given on the Week 3 task uses "Ui" so it should be safer to use it.

PangKuangWei commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,205 @@
+     * @return the request type.
+     * @throws DukeException when the input is invalid.
+     */
+    public static RequestType checkRequest(String request) throws DukeException {
+        if (request.equals("bye")) {
+            return RequestType.EXIT;
+        } else if (request.equals("list")) {
+            return RequestType.LIST;
+        } else if (request.matches("mark \\d+")) {
+            return RequestType.DONE;
+        } else if (request.matches("unmark \\d+")) {
+            return RequestType.ONGOING;
+        } else if (request.matches("delete \\d+")) {
+            return RequestType.DELETE;
+        } else if (request.matches("(?i)^(todo)(.*)")) {
+            return RequestType.TODO;
+        } else if (request.matches("(?i)^(find)(.*)")) {
+            return RequestType.FIND;
+        } else if (request.matches("(?i)^(deadline)(.*)")) {
+            return RequestType.DEADLINE;
+        } else if (request.matches("(?i)^(event)(.*)")) {
+            return RequestType.EVENT;
+        } else {
+            throw new InvalidInputException();
+        }
+    }
```
I like your usage of the matches method with regex. If the functionalities expand you could consider using a switch for the single word commands that you use the equals method for and the remaining if else stack in the default case.

PangKuangWei commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,49 @@
+     */
+    public void run() {
+        ui.greet();
+        boolean isExit = false;
+        while (!isExit) {
```
Perhaps changing `isExit` to "canExit" would be more suitable?

PangKuangWei reviewed (commented):
Your code looks good! I feel that your code was easy to read as it was very organised. It was very hard to find things to comment about.

lilythchu committed:
improve layout

DarrenCsAcc commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,37 @@
+/**
+ * Representation of the find command.
+ */
+public class FindCommand extends Command {
+    private final String keyword;
```
Good use of access modifiers

lilythchu committed:
Handle queries

Duke reply exactly what user's input.

Handling some specific queries makes app more useful.

Parsing the user input to identify which command it belongs to
and returning a response message for that command.

Parsing input is preferable since it utilizes the previous code.

lilythchu committed:
Use Assertions

Classes do not have any checks for their invariants or
assumptions made.

Not checking for these invariants or assumptions will
cause the program to crash or give incorrect results

Add assertions to check to invariants.

Assertions is preferred as it gets the job done and it
can be turned on or off.

lilythchu committed:
Merge pull request #1 from lilythchu/branch-A-Assertions

Use Assertions

lilythchu committed:
Improve Code Quality

Code Quality can be improved

Poor code quality can lead to the code being messy and hard
to understand.

Follow coding standard of CS2103T.

Easy for other people to read and understand my code.

lilythchu committed:
Merge pull request #2 from lilythchu/branch-A-CodeQuality

Improve Code Quality

lilythchu committed:
Set up CI

Project doesn't have CI.

CI enables teams to release a constant flow of software updates
into production to quicken release cycles, low costs, and
reduce the risks associated with development.

Adding a simple CI to run test case when pushing code and
pull request are triggered.

To ensure other's code don't breaking UI.

lilythchu committed:
Fix a typo

lilythchu committed:
Fix checkstyle violations

lilythchu committed:
Fix paths in runtest file

lilythchu committed:
Remove redundant code

lilythchu committed:
Change indentation style

lilythchu committed:
Merge pull request #3 from lilythchu/branch-Setup-CI

Set up CI

lilythchu committed:
Fix wrap text

lilythchu committed:
Add help command

lilythchu committed:
Merge pull request #4 from lilythchu/branch-C-Help

Add help command

lilythchu committed:
Improve UI and add emoji

lilythchu committed:
Merge pull request #6 from lilythchu/branch-A-BetterGui

Improve UI and add emoji

lilythchu committed:
Improve layout

lilythchu committed:
Write User Guide

lilythchu committed:
Update UG

lilythchu committed:
More error handling

lilythchu committed:
Save data automatically

lilythchu committed:
Fix save data bug

lilythchu committed:
Handle empty task list

lilythchu committed:
Update README.md

lilythchu committed:
Update README.md
